src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests_ListPatterns.cs

```diff
@@ -0,0 +1,1159 @@
+
+            N(SyntaxKind.IsPatternExpression);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "c");
+                }
+                N(SyntaxKind.IsKeyword);
+                N(SyntaxKind.RecursivePattern);
+                {
+                    N(SyntaxKind.ListPatternClause);
+                    {
+                        N(SyntaxKind.OpenBraceToken);
+                        N(SyntaxKind.Subpattern);
+                        {
+                            N(SyntaxKind.OrPattern);
+                            {
+                                N(SyntaxKind.SlicePattern);
+                                {
+                                    N(SyntaxKind.DotDotToken);
+                                    N(SyntaxKind.ConstantPattern);
+                                    {
+                                        N(SyntaxKind.IdentifierName);
+                                        {
+                                            N(SyntaxKind.IdentifierToken, "p");
+                                        }
+                                    }
+                                }
+                                N(SyntaxKind.OrKeyword);
+                                N(SyntaxKind.SlicePattern);
+                                {
+                                    N(SyntaxKind.DotDotToken);
+                                    N(SyntaxKind.ConstantPattern);
+                                    {
+                                        N(SyntaxKind.IdentifierName);
+                                        {
+                                            N(SyntaxKind.IdentifierToken, "q");
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        N(SyntaxKind.CloseBraceToken);
+                    }
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void SlicePattern_09()
+        {
+            UsingExpression(@"c is .. var x");
+
+            N(SyntaxKind.IsPatternExpression);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "c");
+                }
+                N(SyntaxKind.IsKeyword);
+                N(SyntaxKind.SlicePattern);
+                {
+                    N(SyntaxKind.DotDotToken);
+                    N(SyntaxKind.VarPattern);
+                    {
+                        N(SyntaxKind.VarKeyword);
+                        N(SyntaxKind.SingleVariableDesignation);
+                        {
+                            N(SyntaxKind.IdentifierToken, "x");
+                        }
+                    }
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void SlicePattern_10()
```

Comment:
It would be good to have variations on these slice tests that have the `..` after the other pattern component. Also, some pattern combinations with positional and length patterns would be good.

Reply:
Added a few tests. Let me know if you have any specific scenario that is not covered.

Reply:
I'd expect to see some of the previous variations you had here. For example, `c is { {}.. }` from `SlicePattern_05`. I'd go through your existing tests and add flipped versions of these.

---
In reply to: [608770979](https://github.com/dotnet/roslyn/pull/51299#discussion_r608770979) [](ancestors = 608770979)

Reply:
Done.